src/Sarif.Multitool.Library/ApplyPolicyCommand.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    public class ApplyPolicyCommand : CommandBase
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public ApplyPolicyCommand(IFileSystem fileSystem = null)
+        {
+            _fileSystem = fileSystem ?? FileSystem.Instance;
+        }
+
+        public int Run(ApplyPolicyOptions applyPolicyOptions)
+        {
+            try
+            {
+                Console.WriteLine($"Applying policy '{applyPolicyOptions.InputFilePath}' => '{applyPolicyOptions.OutputFilePath}'...");
+                Stopwatch w = Stopwatch.StartNew();
+
+                bool valid = ValidateOptions(applyPolicyOptions);
+                if (!valid) { return FAILURE; }
+
+                SarifLog actualLog = ReadSarifFile<SarifLog>(_fileSystem, applyPolicyOptions.InputFilePath);
+
+                actualLog.ApplyPolicies();
+
+                string fileName = CommandUtilities.GetTransformedOutputFileName(applyPolicyOptions);
+
+                Formatting formatting = applyPolicyOptions.PrettyPrint
+                    ? Formatting.Indented
+                    : Formatting.None;
```

Comment:
You know, we have this transformation from "common options" to Newtonsoft Formatting in many places. This could be an extension method on CommonOptions (it would be called like this: `applyPolicyOptions.GetFormatting()`. Ask @michaelcfanning if this is a good time to do that.

Reply:
I think we can do a separate pr for that. So i can tackle all places, what do you think?

---
In reply to: [506574244](https://github.com/microsoft/sarif-sdk/pull/2118#discussion_r506574244) [](ancestors = 506574244)

Reply:
Sure, that's fine.

---
In reply to: [506659476](https://github.com/microsoft/sarif-sdk/pull/2118#discussion_r506659476) [](ancestors = 506659476,506574244)

Reply:
The new pr with this: https://github.com/microsoft/sarif-sdk/pull/2121